toponetx/algorithms/components.py

```diff
@@ -0,0 +1,201 @@
+"""Module to compute connected components on topological domains."""
+from collections.abc import Collection, Hashable, Iterable, Iterator
+from warnings import warn
+
+import networkx as nx
+import numpy as np
+
+from toponetx.classes.cell import Cell
+from toponetx.classes.cell_complex import CellComplex
+from toponetx.classes.colored_hypergraph import ColoredHyperGraph
+from toponetx.classes.combinatorial_complex import CombinatorialComplex
+from toponetx.classes.simplicial_complex import SimplicialComplex
+from toponetx.exception import TopoNetXError
+
+__all__ = [
+    "s_connected_components",
+    "s_component_subcomplexes",
+    "connected_components",
+    "connected_component_subcomplexes",
+]
+
+
+def s_connected_components(
+    cplex, s: int = 1, cells: bool = True, return_singletons: bool = False
```

Comment:
Our API does not use `cplex` but `complex`. Update? Here and everywhere.

Reply:
done